src/NumericalAlgorithms/Integration/GslQuadAdaptive.hpp

```diff
@@ -150,43 +111,51 @@ class GslQuadAdaptive;
  * functions, whereas a lower-order rule saves time for functions with local
  * difficulties, such as discontinuities.
  */
-template <typename... Args>
-class GslQuadAdaptive<IntegralType::StandardGaussKronrod, Args...>
-    : public detail::GslQuadAdaptiveImpl<Args...> {
+template <>
+class GslQuadAdaptive<GslIntegralType::StandardGaussKronrod>
+    : public detail::GslQuadAdaptiveImpl {
  public:
-  using detail::GslQuadAdaptiveImpl<Args...>::GslQuadAdaptiveImpl;
-  Result operator()(double lower_boundary, double upper_boundary,
-                    double tolerance_abs, int key,
-                    double tolerance_rel = 0.) const noexcept {
-    Result result;
-    int status = gsl_integration_qag(
-        &(this->integrand_), lower_boundary, upper_boundary, tolerance_abs,
-        tolerance_rel, this->max_intervals_, key, this->workspace_.get(),
-        &result.value, &result.error);
+  using detail::GslQuadAdaptiveImpl::GslQuadAdaptiveImpl;
+  template <typename IntegrandType>
+  double operator()(IntegrandType&& integrand, double lower_boundary,
+                    double upper_boundary, double tolerance_abs, int key,
+                    double tolerance_rel = 0.) const {
```

Comment:
make all arguments `const` (instead of `integrand`). same for the other integral types